### PR TITLE
PAT-1883 DCS - performance of dashboard setting off database CPU alerts

### PR DESCRIPTION
**Pull request by steverendell**
Rewritten SQL to improve efficiency

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '3.1'
+version: '3.8'
 services:
```
Comment (steverendell) [inline docker-compose-test.yml:2]: added useful stuff to this docker-compose

File: server/data/statsClient.js [hunk 2]
```diff
@@ -34,22 +34,31 @@ module.exports = {
...
       text: `
         with cat_table as (
           select booking_id,
                  sequence_no,
-                 coalesce (form_response -> 'supervisor' ->'review' ->>'supervisorOverriddenCategory',
-                           form_response -> 'recat' -> 'decision' ->>'category') as cat
-          from form)
-        select count(*),
-          (select cat from cat_table where booking_id = f.booking_id and sequence_no = (select max(f2.sequence_no) - 1 from form f2 where f2.booking_id = f.booking_id)) as previous,
-          (select cat from cat_table where booking_id = f.booking_id and sequence_no = (select max(f2.sequence_no) from form f2 where f2.booking_id = f.booking_id)) as current
```
Comment (steverendell) [inline server/data/statsClient.js:48]: these 2 subselects werer really slow

File: server/data/statsClient.js [hunk 3]
```diff
@@ -34,22 +34,31 @@ module.exports = {
...
+                 prison_id,
+                 coalesce(form_response -> 'supervisor' -> 'review' ->> 'supervisorOverriddenCategory',
+                          form_response -> 'recat' -> 'decision' ->> 'category') as cat
+          from form
+          where status = 'APPROVED'
+            and cat_type = 'RECAT'
+            and ($1::date is null or approval_date <= $1::date)
+        ),
+             arrays_table as (
+               select array_agg(array [prison_id,cat] order by sequence_no desc) as data
```
Comment (steverendell) [inline server/data/statsClient.js:53]: first 2 array elements are current and previous

**Review by lbennett-moj — COMMENTED** [review]
LGTM!

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**